### PR TITLE
made the emoji store getByName method sync

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
made the emoji store getByName method sync

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11299

### CODE REVIEW

File: app/emoji.go [hunk 1]
```diff
@@ -190,11 +190,11 @@ func (a *App) GetEmojiByName(emojiName string) (*model.Emoji, *model.AppError) {
 		return nil, model.NewAppError("GetEmoji", "api.emoji.storage.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	result := <-a.Srv.Store.Emoji().GetByName(emojiName)
```
Comment (jespino) [inline app/emoji.go]: You can just `return a.Srv.Store.Emoji().GetByName(emojiName)` here

File: app/import_functions_test.go [hunk 2]
```diff
@@ -2681,8 +2681,8 @@ func TestImportImportEmoji(t *testing.T) {
 	err := th.App.ImportEmoji(&data, true)
 	assert.NotNil(t, err, "Invalid emoji should have failed dry run")
 
-	result := <-th.App.Srv.Store.Emoji().GetByName(*data.Name)
-	assert.Nil(t, result.Data, "Emoji should not have been imported")
+	emoji, err := th.App.Srv.Store.Emoji().GetByName(*data.Name)
```
Comment (jespino) [inline app/import_functions_test.go]: If you are ignoring the error just use `_` to be explicit about it. Other option (I think is better) is to use `require.NotNil(t, err)` or `require.Nil(t, err)` depending of what you really expect there.

File: app/import_functions_test.go [hunk 3]
```diff
@@ -2700,8 +2700,8 @@ func TestImportImportEmoji(t *testing.T) {
 	err = th.App.ImportEmoji(&data, false)
 	assert.Nil(t, err, "Valid emoji should have succeeded apply mode")
 
-	result = <-th.App.Srv.Store.Emoji().GetByName(*data.Name)
-	assert.NotNil(t, result.Data, "Emoji should have been imported")
+	emoji, err = th.App.Srv.Store.Emoji().GetByName(*data.Name)
```
Comment (jespino) [inline app/import_functions_test.go]: same here.

File: store/sqlstore/emoji_store.go [hunk 4]
```diff
@@ -100,26 +100,27 @@ func (es SqlEmojiStore) Get(id string, allowFromCache bool) (*model.Emoji, *mode
...
-		var emoji *model.Emoji
+func (es SqlEmojiStore) GetByName(name string) (*model.Emoji, *model.AppError) {
 
-		if err := es.GetReplica().SelectOne(&emoji,
-			`SELECT
+	var emoji *model.Emoji
+
+	if err := es.GetReplica().SelectOne(&emoji,
+		`SELECT
 				*
```
Comment (jespino) [inline store/sqlstore/emoji_store.go]: Please, keep the consistency of the indentation in the query string (just remove 1 tab on each line)

File: store/sqlstore/emoji_store.go [hunk 5]
```diff
@@ -100,26 +100,27 @@ func (es SqlEmojiStore) Get(id string, allowFromCache bool) (*model.Emoji, *mode
...
 				Name = :Name
 				AND DeleteAt = 0`, map[string]interface{}{"Name": name}); err != nil {
-			result.Err = model.NewAppError("SqlEmojiStore.GetByName", "store.sql_emoji.get_by_name.app_error", nil, "name="+name+", "+err.Error(), http.StatusInternalServerError)
-			if err == sql.ErrNoRows {
-				result.Err.StatusCode = http.StatusNotFound
-			}
-		} else {
-			result.Data = emoji
+
+		if err == sql.ErrNoRows {
```
Comment (jespino) [inline store/sqlstore/emoji_store.go]: Nice rewrite :heart:

File: store/sqlstore/emoji_store.go [hunk 6]
```diff
@@ -105,13 +105,13 @@ func (es SqlEmojiStore) GetByName(name string) (*model.Emoji, *model.AppError) {
...
 
 	if err := es.GetReplica().SelectOne(&emoji,
-		`SELECT
-				*
-			FROM
-				Emoji
-			WHERE
-				Name = :Name
-				AND DeleteAt = 0`, map[string]interface{}{"Name": name}); err != nil {
+	`SELECT
```
Comment (jespino) [inline store/sqlstore/emoji_store.go]: This exact line was correctly placed, was the other lines of the string that were shifted.

**Comment by pradeepmurugesan** [comment]
> A tiny extra detail.

Yes.. Ran a gofmt on the file. :-) so that I don't mess it up again.. Hopefully it passes the build now

**Comment by jespino** [comment]
@pradeepmurugesan Please, avoid `push -f` we always squash and merge the PRs so the result will be the same, one single commit, but now I have to review again the whole PR. In this case is not a big deal because is a small PR, but is a good policy to follow.

**Comment by pradeepmurugesan** [comment]
> @pradeepmurugesan Please, avoid `push -f` we always squash and merge the PRs so the result will be the same, one single commit, but now I have to review again the whole PR. In this case is not a big deal because is a small PR, but is a good policy to follow.

Sure.. 👍

**Comment by hanzei** [comment]
Thanks @pradeepmurugesan :+1:

**Pull request merged**